pkg/kubectl/cmd/util/printing.go

```diff
@@ -126,6 +127,15 @@ func PrinterForCommand(cmd *cobra.Command, outputOpts *printers.OutputOptions, m
 		return nil, err
 	}
 
+	// if printer.isGeneric() not true, it must be a HumanReadablePrinter
+	// we execute AddHandlers() here before maybeWrapSortingPrinter so that we don't
+	// need to use delegatePrinter convert again outsider when it's a SortingPrinter
+	if !printer.IsGeneric() {
+		if humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter); ok {
```

Comment:
I had a thinking about keep only one `if`, but I think we should check printer.IsGeneric() first, then do that convert, do you mean this way: 
```
if humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter); ok && !printer.IsGeneric(){
	printersinternal.AddHandlers(humanReadablePrinter)
}
```
or this way ?
```
if !printer.IsGeneric() {
          humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter);
          printersinternal.AddHandlers(humanReadablePrinter)
}
```